mud-contracts/world/src/modules/smart-gate/systems/SmartGateSystem.sol

```diff
@@ -262,6 +286,22 @@ contract SmartGateSystem is EveSystem, AccessModified {
     return distanceSquaredMeters <= (maxDistance * maxDistance);
   }
 
+  /**
+   * @notice delete the existing link between the source and destination gate
+   * @param sourceGateId is the smartObjectId of the source gate
+   */
+  function _deleteExistingLink(uint256 sourceGateId) internal {
```

Comment:
How is this different from unlink? should we be doing this logic in there? 
`if (!linkData.isLinked) {` indicates that we have an unlinked gate with dangling links that need to be deleted

Reply:
yes you are right. Its ideal to delete the dangling ref by deleting the record during unlink. But
The external services are subscribed to `__lastUpdatedBlockNumber` which is a column automatically updated when the record is updated.  Based on the change in the `__lastUpdatedBlockNumber` they inform the game that the gates are unlinked. 

So if we delete the record, then there is no `__lastUpdatedBlockNumber` for that record. So there is no way they can receive a event to inform the game. So when they link we leave the record in the table by marking `isLinked` false so they get updates. 

Later if there is a new link with anyone of the existing gates, then we delete the record.

Reply:
I will add some comments and change the description for more clarity